service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java

```diff
@@ -280,4 +283,45 @@ private ApiDeleteAzureLandingZoneJobResult toApiDeleteAzureLandingZoneJobResult(
     }
     return apiJobResult;
   }
+
+  public ApiAzureLandingZone getAzureLandingZone(BearerToken bearerToken, UUID landingZoneId) {
+    features.azureEnabledCheck();
+    LandingZone landingZoneRecord = landingZoneService.getLandingZone(bearerToken, landingZoneId);
+    return toApiAzureLandingZone(landingZoneRecord);
+  }
+
+  public ApiAzureLandingZoneList listAzureLandingZones(
+      BearerToken bearerToken, UUID billingProfileId) {
+    features.azureEnabledCheck();
+    if (billingProfileId != null) {
+      return getAzureLandingZonesByBillingProfile(bearerToken, billingProfileId);
+    }
+    List<LandingZone> landingZones = landingZoneService.listLandingZones(bearerToken);
+    return new ApiAzureLandingZoneList()
+        .landingzones(
+            landingZones.stream().map(this::toApiAzureLandingZone).collect(Collectors.toList()));
+  }
+
+  private ApiAzureLandingZoneList getAzureLandingZonesByBillingProfile(
+      BearerToken bearerToken, UUID billingProfileId) {
+    ApiAzureLandingZoneList result = new ApiAzureLandingZoneList();
+    List<LandingZone> landingZones =
+        landingZoneService.getLandingZonesByBillingProfile(bearerToken, billingProfileId);
+    if (landingZones.size() > 0) {
+      // The enforced logic is 1:1 relation between Billing Profile and a Landing Zone.
+      // The landing zone service returns one record in the list if landing zone exists
+      // for a given billing profile.
+      result.addLandingzonesItem(toApiAzureLandingZone(landingZones.get(0)));
+    }
+    return result;
```

Comment:
Since we are returning either 0 or 1 landing zones, should we return an Optional result instead of a list? I see right now this is sharing the API with listLandingZones, and I do think it saves complexity this way, so I'm ok with that

Reply:
I think using List as the return type is fine -- we enforce 1:1 for now but this may change in the future to a 1:many model.